src/app/weather-data/weather-data.component.html

```diff
@@ -0,0 +1,9 @@
+
+<div *ngIf="data.length==3">
+  <div *ngFor="let d of data;index as i;" >
```

Comment:
rather than doing the loop here and referencing elements in the array, it'd be better to do the loop in the app component, rendering this one out 3 times & only passing the relevant data for each individual location in

it'll keep this one much simpler and make it considerably easier to test